File: Scripts/Player/Movement/PlayerStats.cs
/*
Contains and updates the players stats
By: Nick Petruccelli
Last Modified: 08/19/2023
*/

using System;
using System.Collections.Generic;
using UnityEngine;

public class PlayerStats : MonoBehaviour
{
	
    public float maxHealth = 10f;
    public float curHealth = 10f;
    public float damageMult = 1f;
    public float critChance = 0;
    public float critMultiplier = 1;
    public ParticleSystem deathEffect;
    public List<string> equippedItems;
    public HealthBar healthBar;
    
    
    // Not going to lie some SerializeField went away this is what I think it is check it
    [SerializeField] private LayerMask layerMask;

    private void Start()
    {
        healthBar.SetHealth(curHealth);
    }

    public void TakeDamage(float damage)
    {
        curHealth -= damage;

        healthBar.SetHealth(curHealth/maxHealth);

        if (curHealth <= 0f)
        {
            Die();
        }
    }
    
    private void Die()
    {
        Instantiate(deathEffect, transform.position, Quaternion.identity);
        Destroy(gameObject);
    }
}

File: Scripts/Player/Movement/player_stats.py
"""
Contains and updates the players stats
By: Nick Petruccelli
Last Modified: 08/19/2023
"""


class PlayerStats:

    maxHealth: float = 10.0
    curHealth: float = 10.0
    damageMult: float = 1.0
    critChance: float = 0.0
    critMultiplier: float = 1.0

    def __init__(self):
        # Once healthbar is implemented, set healthbar health
        # healthBar.SetHealth(curHealth);
        ...

    # TODO take damage should be a player based item that inflicts it player stat dataclass
    def take_damage(self, damage: float):
        self.curHealth -= damage

        # healthBar.SetHealth(curHealth / maxHealth);

        if self.curHealth < 0:
            self.kill_player()

    # TODO same is for the kill player command
    def kill_player(self):
        ...
